src/tools/manifest2proto.ts

```diff
@@ -205,15 +206,30 @@ function checkExpressionToProtoPayload(
         case CheckType.IsFromHandle:
         case CheckType.IsFromOutput:
         case CheckType.IsFromStore:
-          // TODO(bgogul):
-          return null;
+          throw new Error(`Unsupported CheckType for check: ${JSON.stringify(condition)}.`);
         default:
-          throw new Error('Unknown CheckType');
+          throw new Error(`Unknown CheckType for check: ${JSON.stringify(condition)}.`);
       }
     }
   }
 }
 
+/** Constructs an AccessPathProto payload. */
+function accessPathProtoPayload(
+    spec: ParticleSpec,
+    connectionSpec: HandleConnectionSpec,
+    fieldPath: string[]
+) {
+  const accessPath: {particleSpec: string, handleConnection: string, selectors?: {field: string}[]} = {
+    particleSpec: spec.name,
+    handleConnection: connectionSpec.name
+  };
+  if (fieldPath.length) {
```

Comment:
Just a clarification: do we need this check? Won't `fieldPath.map` take care of this case?

Reply:
`fieldPath.map()` will return an empty list if `fieldPath` is empty, which means we will set `accessPath` to the empty list (rather than just not setting the property. 

Functionally those two options are equivalent, but it makes unit tests slightly more annoying, since we now have to add in `selectors: []` everywhere they're not used.

Reply:
Sounds good. :)